app/jobs/scheduled/poll_feed.rb

```diff
@@ -79,35 +76,43 @@ def topics
       private
 
       def fetch_rss
-        SimpleRSS.parse open(@feed_url, allow_redirections: :all)
-      rescue OpenURI::HTTPError, SimpleRSSError
+        if SiteSetting.embed_username_key_from_feed.present?
+          FeedElementInstaller.install_rss_element(SiteSetting.embed_username_key_from_feed)
+          FeedElementInstaller.install_atom_element(SiteSetting.embed_username_key_from_feed)
+        end
+
+        RSS::Parser.parse(open(@feed_url, allow_redirections: :all), false)
```

Comment:
This feels a bit unsafe due to infinite redirection potential, should probably use final destination here to be consistent and probably excon to download (we do want a handover method on final destination so maybe add that as well)

Reply:
> should probably use final destination here to be consistent and probably excon to download

Didn't know about the latter, will do!

> a handover method on final destination

Could you explain this a bit more? Specifically what is a handover method and what is it expected to do?

Reply:
specifically final destination is a bit wasteful cause it does a HEAD only to do a GET afterwards to get the actual data, in this case a single GET would do, so it would be nice if you could use it to handover the "open" GET request and avoid the HEAD req.

Reply:
Thanks for taking the time to explain that, but I am still a bit confused 😰 on the following point  

> in this case a single GET would do

The proposed flow to resolve the potential infinite redirection is to: (1) get the final destination of the feed url (with HEAD, internally), and then (2) `excon` GET's the resolved url.

The HEAD request is unneeded when there are no (infinite) redirection, of which we cannot be sure without FinalDestination (that sends HEADs). What am I missing here?

Reply:
Thing is FinalDestination could just use GET and hand over the get something like this could work

```
txt = FinalDestination.new(url).get
```

Then it can internally be smart enough to do GETs instead of HEAD

This optimises for the general case where a single GET is good enough to find the final destination, redirects are pretty much the same cost be they HEAD or GET so overall cost is unconditionally cheaper.

Reply:
I am scope creeping here, so for starters just implement final destination as it is implemented elsewhere, I can merge it in and then we can do another pr for any final destination optimisations.